File: sdks/python/examples/simple/bulk.py
import asyncio
# ❓ Running a Task
from examples.simple.worker import SimpleInput, step1

step1.run(SimpleInput(message="Hello, World!"))
# !!

async def main():
    # ❓ Bulk Run a Task
    greetings = ["Hello, World!", "Hello, Moon!", "Hello, Mars!"]
    
    results = await step1.aio_run_many(
        [
            # run each greeting as a task in parallel
            step1.create_bulk_run_item(
                input=SimpleInput(message=greeting),
            ) for greeting in greetings
        ]
    )

    # this will await all results and return a list of results
    print(results)
    # !!

    # ❓ Running Multiple Tasks
    result1 = step1.aio_run(SimpleInput(message="Hello, World!"))
    result2 = step1.aio_run(SimpleInput(message="Hello, Moon!"))

    #  gather the results of the two tasks
    results = await asyncio.gather(result1, result2)

    #  print the results of the two tasks
    print(results[0].TransformedMessage)
    print(results[1].TransformedMessage)
    # !!

File: sdks/typescript/src/v1/examples/simple/bulk.ts
/* eslint-disable no-console */
import { hatchet } from '../hatchet-client';
import { simple, SimpleInput } from './workflow';

async function main() {
  // > Bulk Run a Task
  const res = await simple.run([
    {
      Message: 'HeLlO WoRlD',
    },
    {
      Message: 'Hello MoOn',
    },
  ]);

  // 👀 Access the results of the Task
  console.log(res[0].TransformedMessage);
  console.log(res[1].TransformedMessage);
  // !!

  // > Bulk Run Tasks from within a Task
  const parent = hatchet.task({
    name: 'simple',
    fn: async (input: SimpleInput, ctx) => {
      // Bulk run two tasks in parallel
      const child = await ctx.bulkRunChildren([
        {
          workflow: simple,
          input: {
            Message: 'Hello, World!',
          },
        },
        {
          workflow: simple,
          input: {
            Message: 'Hello, Moon!',
          },
        },
      ]);

      return {
        TransformedMessage: `${child[0].TransformedMessage} ${child[1].TransformedMessage}`,
      };
    },
  });
  // !!
}

if (require.main === module) {
  main();
}
